### PR TITLE
Simplified email validation Regexp

### PR DESCRIPTION
**Pull request by en3on**

### CODE REVIEW

File: controllers/auth-controller.js [hunk 1]
```diff
@@ -10,32 +22,45 @@ async function register(req, res, next) {
         const query = await User.findOne({email});
 
         if (query === null) {
-          const user = await generateUser(firstName, lastName, email, password);
+          const user = await generateUser({
+            firstName,
+            lastName,
+            email,
+            password,
+          });
```
Comment (PraveenMut) [inline controllers/auth-controller.js:30]: This looks so much better, nice work!

File: controllers/auth-controller.js [hunk 2]
```diff
@@ -10,32 +22,45 @@ async function register(req, res, next) {
...
           next(new ValidationError(400, 'Email already in use!'));
         }
       } catch (err) {
         next(err);
       };
     } else {
-      next(new ValidationError(400, 'Please ensure all fields are filled out!'));
+      next(
+          new ValidationError(400, 'Please ensure all fields are filled out!')
+      );
```
Comment (PraveenMut) [inline controllers/auth-controller.js:47]: mmmmm, this much more readable!

File: controllers/auth-controller.js [hunk 3]
```diff
@@ -10,32 +22,45 @@ async function register(req, res, next) {
...
   };
 };
 
+/**
+ * login function for users
+ * @param {Object} req express request object
+ * @param {Object} res express response object
+ * @param {Function} next express next function
+ * @return {Object} returns 200 status with token
+ */
```
Comment (PraveenMut) [inline controllers/auth-controller.js:60]: loving that jsdoc goodness!

File: utils/auth-utils.js [hunk 4]
```diff
@@ -22,22 +38,33 @@ async function comparePassword(password, hash) {
...
   }
 }
-
+/**
+ * validates email against regexp
+ * @param {String} email email address
+ */
 function validateEmail(email) {
-  const regexp = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;
+  const regexp = /\b[\w.%-]+@[\w.-]+.[a-z]{2,}\b/;
```
Comment (PraveenMut) [inline utils/auth-utils.js]: WOW, now that's an improvement! However, I did find a slight bug while testing this out in a regexr tester website (https://regexr.com). It seems to prematurely match the email without a domain. For example, if I type in john.snow@somedomain => this results in a match, whereas the other (albeit quite a bit a more uglier), results in a no match. https://imgur.com/a/yEujx9P

Comment (PraveenMut) [reply utils/auth-utils.js]: Probably not an issue, but just in case.

**Review by PraveenMut — CHANGES_REQUESTED** [review]
Everything looks amazing as usual!!!, just brought up a slight regex issue. Loving the mass semicolon fixes :)

File: utils/auth-utils.js (hunk 4 above)
Comment (en3on) [reply utils/auth-utils.js]: Thanks for catching that man! That's a big bug haha! I just left out a backslash :stuck_out_tongue_closed_eyes:

Comment (PraveenMut) [reply utils/auth-utils.js]: ahhh, sweet! Yeah thought was just a simple typo!

File: utils/auth-utils.js [hunk 5]
```diff
@@ -22,22 +38,33 @@ async function comparePassword(password, hash) {
...
   }
 }
-
+/**
+ * validates email against regexp
+ * @param {String} email email address
+ */
 function validateEmail(email) {
-  const regexp = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;
+  const regexp = /\b[\w.%-]+@[\w.-]+\.[a-z]{2,}\b/;
```
Comment (PraveenMut) [inline utils/auth-utils.js:67]: PERFECT!! This works like a charm! Awesome man!

File: test/unit-tests/Users/login.js [hunk 6]
```diff
@@ -10,14 +9,16 @@ const User = require('../../../models/User.js');
...
 describe('POST /auth/login', () => {
   beforeEach(async () => {
     api = require('../../../api.js');
     url = '/auth/login';
     user = {
-      _id: mongoose.Types.ObjectId(),
+      _id: new mongoose.Types.ObjectId(),
       firstName: 'Bill',
       lastName: 'Clinton',
       email: 'billc@whitehouse.com',
```
Comment (PraveenMut) [inline test/unit-tests/Users/login.js:24]: 10/10 stub lol

**Review by PraveenMut — APPROVED** [review]
Super nice man! The regex now works like a champ, cool 💯! Question, do you leave dangling commas in arrays too?

**Review by alex-double-u-old — APPROVED** [review]
this is the work of a god

**Pull request merged**